### ISSUE TITLE
Clarify why incremental + runtime builds are prohibited

### ISSUE THREAD

**Issue by asottile** [openshift/source-to-image#824]
I'm reaching for something like this but [the documentation](https://github.com/openshift/source-to-image/blob/master/docs/runtime_image.md#extended-build-and-incremental-build) indicates this is prohibited.

I'm hoping to understand more of the _why_ here.  Is it not possible given the current architecture or just not implemented?

**Comment by bparees** [comment]
not implemented, but not likely to be implemented as the extended build feature was ultimately deemed to be an unnecessary re-invention of capabilities that exist elsewhere.

for example, docker multistage builds allow you to extract content from one image and layer it into another:
https://docs.docker.com/engine/userguide/eng-image/multistage-build/#use-multi-stage-builds

and openshift chained builds do the same, more generically:
https://docs.openshift.org/latest/dev_guide/builds/advanced_build_operations.html#dev-guide-chaining-builds

So basically we'd recommend you do a standard s2i build(which could be an incremental build) to compile your artifacts, and then extract them from your image and layer them into your runtime image.  It's a bit more manual steps on your side, but against the complexity of the implementation in s2i and forcing users to implement a runtime-assemble script, etc, it was deemed to be a dead end.

**Comment by asottile** [comment]
Sounds great -- for others looking for something like this I'm essentially using these steps:

```bash
# run the initial s2i build
s2i build ...

# create a "container" from that s2i built image, we'll copy things out of it
build_cid=$(docker create $resulting_built_image)
docker cp $build_cid:/path tmp/path
docker rm $build_cid

# create a container of your runtime base image, we'll copy into it
run_cid=$(docker create $runtime_image)
docker cp tmp/path $run_cid:/path
runtime_image_id=$(docker commit $run_cid)
docker rm $run_cid

# tag the committed layer as something meaningful
docker tag $runtime_image_id my-tag
```
